extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandlerBase.java

```diff
@@ -41,19 +45,43 @@ private void doHandle(RoutingContext ctx) {
             Arc.container().instance(CurrentIdentityAssociation.class).get().setIdentity(user.getSecurityIdentity());
         }
         SmallRyeHealthReporter reporter = Arc.container().instance(SmallRyeHealthReporter.class).get();
-        SmallRyeHealth health = getHealth(reporter, ctx);
-        HttpServerResponse resp = ctx.response();
-        if (health.isDown()) {
-            resp.setStatusCode(503);
+        getHealth(reporter, ctx).subscribe().with(health -> {
```

Comment:
But I don't understand what exactly do you need from SR Health? In SR Health we don't have Quarkus concepts of worker and eventloop threads. It just returns Uni that can be handled accordingly. So all that is actually required should be to put here something like
```suggestion
        getHealth(reporter, ctx).runSubscriptionOn(MutinyHelper.executor(vertx)).subscribe().with(health -> {
```

Reply:
If @ahus1 does this and re-runs my analysis and concludes that my points are addressed, it's fine with me

Reply:
Sure, will have a look. Stay tuned...

Reply:
I tested it with a small example app, and debugging both SYNC, ASYNC and a mix SYNC/ASYNC checks. Here are the results: 

* I added `.emitOn(MutinyHelper.executor(ctx.vertx()))`, the `subscribe().with()` in `SmallRyeHealthHandlerBase`, the code to prepare the response always ran on the Vert.x event loop. 
* The code in `SmallRypeHealthReporter#getHealthAsync` which calls `Uni.combine().all().unis(healthCheckUnis).combinedWith(responses -> { ... }`) which calculates the status of the response will run on the thread of the last completed `Uni` - either SYNC or ASYNC. It is calculates the global outcome with the local data that was previously been collected, which can IMHO run on any kind of thread. See https://github.com/smallrye/smallrye-health/blob/7d40f8f85e4a8cae43f7c1b2cc3b8d72a2dfa965/implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java#L510-L521

I also tried to add `runSubscriptionOn()` in `SmallRyeHealthHandlerBase`, but didn't have any effect. I'm new to Mutiny, still it seems to me that `combinedWith` at least not in the way it is used in `SmallRypeHealthReporter` works with that option.